components/Opportunities.tsx

```diff
@@ -0,0 +1,106 @@
+"use client"
+import React, { useState, useEffect } from "react"
+import Link from "next/link"
+import { LinkCard } from "@/components/LinkCard"
+import {
+  MapPinIcon,
+  ArrowRightIcon,
+  UserPlusIcon,
+} from "@heroicons/react/24/solid"
+
+interface OpportunityProps {
+  total: number
+  jobPostings: PositionProps[]
+  facets: Object[]
+  userAuthenticated: boolean
+}
+
+interface PositionProps {
+  title: string
+  externalPath: string
+  locationsText: string
+  postedOn: string
+  bulletFields: string[]
+}
+
+export const Opportunities: React.FC = () => {
+  const initialData = {
+    total: 0,
+    jobPostings: [],
+    userAuthenticated: false,
+    facets: [],
+  }
+  const [data, setData] = useState<OpportunityProps>(initialData)
+  const [isLoading, setLoading] = useState(true)
+  const [error, setError] = useState(null)
+
+  const handleApiCall = async () => {
+    try {
+      const res = await fetch("/api/about", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      })
+
+      if (!res.ok) {
+        throw new Error(`Failed to fetch: ${res.status}`)
+      }
+
+      const opportunities = await res.json()
+      setData(opportunities.json)
+    } catch (err: any) {
+      setError(err.message)
+    } finally {
+      setLoading(false)
+    }
+  }
+  useEffect(() => {
+    handleApiCall()
+  }, [])
```

Comment:
Just out of curiosity, why not handle this in one of the server components? That way you only have to transmit the HTML of the page and you don't have to worry about loading state or whatever.

Reply:
Do you mean pulling this out into its own server component? So "OpportunitiesSection" could be the html and such and Opportunities.tsx could be the server component?

I personally find it frustrating when things get too abstracted and there are 6 files that I need to look into to understand what specifically I need to edit/change; projects get overwhelming as the components folder grows; however, this is running on each page load which is probably overkill? 

Thinking about it, this is worth moving for functionality; I need some help thinking about file structure in the future though!

Reply:
Sortof! This is informed by a convention I've come to for myself. When possible, I like to do all my fetching / data interactions at the highest component level I can / is reasonable. For NextJS, it's on the `page.tsx` level. And I think some developers on Next JS and I are of the same opinion about this because all `page.tsx` components are `async` by default!

So the pattern would be that you do your asynchronous fetch action in the page, `await` the result, and then pass the resolved data to the UI component later on. For this example, it might look something like this:

### `app/about/page.tsx`
```jsx
export default async function About() {
  try {
    // Omitting function definition for brevity
    const data: OpportunityProps = await handleApiCall();
    return <Opportunities data={data} />;
  }
  catch (ex) {
    // Handle the case when the fetch fails for whatever reason, display a different page.
    return <p>{ex}</p>;
  }
}
```

### `components/Opportunities.tsx`
```jsx
"use client";
export function Opportunities({ data }: {data:OpportunityProps}) {
  // However you want to display the fetched data, you can omit the fetch effect because you're getting passed validated data.
  return (
    <>
      {data.map( /* ... */)}
    </>
  );
}
```

Does this make sense? I got a little lazy when writing this 😓